staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -231,6 +229,7 @@ func (p *Parser) parseRecursive(cur *ListNode) error {
 func (p *Parser) parseNumber(cur *ListNode) error {
 	r := p.peek()
 	if r == '+' || r == '-' {
+		//lint:ignore SA4006 skip the rune
 		r = p.next()
```

Comment:
Why not simply say `p.next()` instead of `r = p.next()`? The assignment doesn't do anything.

Reply:
oh...wonderfull. Thanks @dominikh